NetSSL_OpenSSL/include/Poco/Net/NetSSL.h

```diff
@@ -61,10 +61,37 @@
 //
 #if defined(_MSC_VER)
 	#if !defined(POCO_NO_AUTOMATIC_LIBS)
-		#if !defined(POCO_EXTERNAL_OPENSSL)
-			#pragma comment(lib, "libcrypto.lib")
-			#pragma comment(lib, "libssl.lib")
-		#endif // POCO_EXTERNAL_OPENSSL
+		#if defined(POCO_INTERNAL_OPENSSL_MSVC_VER)
```

Comment:
yes, Crypto.h does all the work.

`POCO_INTERNAL_OPENSSL_MSVC_VER` is a new define that comes "automatically" from the [define](https://github.com/pocoproject/openssl/blob/develop/build/include/openssl/opensslv.h#L113) injected by poco internal openssl build.

we issue a compile-time warning if user has defined external ssl, but is using internal headers. if user has not specified anything, internal build is used by default.

Reply:
Good to know: we also base some [hacking](https://github.com/pocoproject/poco/blob/develop/Crypto/src/OpenSSLInitializer.cpp#L43-L54) (in order to prevent static linking errors) on that define